lib/chef/provider/cron.rb

```diff
@@ -220,9 +220,14 @@ def get_crontab_entry
         [ :mailto, :path, :shell, :home ].each do |v|
           newcron << "#{v.to_s.upcase}=\"#{new_resource.send(v)}\"\n" if new_resource.send(v)
         end
+        idempotency = true
         new_resource.environment.each do |name, value|
+          if %w{MAILTO PATH SHELL HOME}.include?(name)
```

Comment:
This is certainly better than what we have today, but it feels like we could just adjust `cron_different?` to compare these better?

Reply:
@vijaymmali1990 I feel this should go in resource.

Reply:
Yes @jaymzh I have done some changes in `cron_different?` and I have also tested these which gives us positive results.

Reply:
@Vasu1105 no we can not add this in resource as for the `environment` variable it does not go in resource file.